File: sidecars/apps-cache/classify.go
package main

import "strings"

func shouldSkipShortcut(name string) bool {
	nameLower := strings.ToLower(name)
	return strings.Contains(nameLower, "uninstall") ||
		strings.Contains(nameLower, "unins") ||
		strings.Contains(nameLower, "setup") ||
		strings.Contains(nameLower, "installer")
}

func categorizeShortcut(name string, shortcutPath string) string {
	nameLower := strings.ToLower(name)
	pathLower := strings.ToLower(shortcutPath)

	if strings.Contains(nameLower, "chrome") ||
		strings.Contains(nameLower, "firefox") ||
		strings.Contains(nameLower, "edge") ||
		strings.Contains(nameLower, "browser") {
		return "browser"
	}

	if strings.Contains(nameLower, "visual studio") ||
		strings.Contains(nameLower, "vscode") ||
		strings.Contains(nameLower, "code") ||
		strings.Contains(nameLower, "git") ||
		strings.Contains(pathLower, "\\microsoft vs code\\") {
		return "development"
	}

	if strings.Contains(nameLower, "word") ||
		strings.Contains(nameLower, "excel") ||
		strings.Contains(nameLower, "powerpoint") ||
		strings.Contains(nameLower, "office") ||
		strings.Contains(nameLower, "wps") {
		return "office"
	}

	if strings.Contains(nameLower, "wechat") ||
		strings.Contains(nameLower, "qq") ||
		strings.Contains(nameLower, "dingtalk") ||
		strings.Contains(nameLower, "teams") ||
		strings.Contains(nameLower, "微信") ||
		strings.Contains(nameLower, "钉钉") {
		return "communication"
	}

	if strings.Contains(nameLower, "player") ||
		strings.Contains(nameLower, "music") ||
		strings.Contains(nameLower, "video") ||
		strings.Contains(nameLower, "photoshop") {
		return "media"
	}

	return "other"
}

File: src-tauri/src/apps_classifier/classify.rs
use std::path::Path;

pub fn should_skip_shortcut_name(name: &str) -> bool {
    let name_lower = name.to_lowercase();
    name_lower.contains("uninstall")
        || name_lower.contains("unins")
        || name_lower.contains("setup")
        || name_lower.contains("installer")
}

pub(super) fn categorize_shortcut(name: &str, shortcut_path: &Path) -> String {
    let name_lower = name.to_lowercase();
    let path_lower = shortcut_path.to_string_lossy().to_lowercase();

    if name_lower.contains("chrome")
        || name_lower.contains("firefox")
        || name_lower.contains("edge")
        || name_lower.contains("browser")
    {
        return "browser".to_string();
    }

    if name_lower.contains("visual studio")
        || name_lower.contains("vscode")
        || name_lower.contains("code")
        || name_lower.contains("git")
        || path_lower.contains("\\microsoft vs code\\")
    {
        return "development".to_string();
    }

    if name_lower.contains("word")
        || name_lower.contains("excel")
        || name_lower.contains("powerpoint")
        || name_lower.contains("office")
        || name_lower.contains("wps")
    {
        return "office".to_string();
    }

    if name_lower.contains("wechat")
        || name_lower.contains("qq")
        || name_lower.contains("dingtalk")
        || name_lower.contains("teams")
        || name_lower.contains("微信")
        || name_lower.contains("钉钉")
    {
        return "communication".to_string();
    }

    if name_lower.contains("player")
        || name_lower.contains("music")
        || name_lower.contains("video")
        || name_lower.contains("photoshop")
    {
        return "media".to_string();
    }

    "other".to_string()
}
